modules/tribes/client/components/CirclePage.component.js

```diff
@@ -0,0 +1,190 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useTranslation } from 'react-i18next';
+
+import * as api from '../api/tribes.api';
+import JoinButton from './JoinButton';
+import { getRouteParams } from '@/modules/core/client/services/angular-compat';
+
+export default function CirclePage({ user, goBack }) {
+  const [circle, setCircle] = useState(null);
+  const { t } = useTranslation('tribes');
+
+  const fetchData = async () => {
+    const circle = await api.get(getRouteParams().circle);
+    setCircle(circle);
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  // Circle loading
+  if (!circle) {
+    return (
+      <div className="container container-spacer text-muted text-center">
```

Comment:
We can use [`LoadingIndicator`](https://github.com/Trustroots/trustroots/blob/061847721413977c17c9623b23c68ed5351a8a0b/modules/core/client/components/LoadingIndicator.js) component here.